wasm/jit/jit_arm64.go

```diff
@@ -25,17 +25,38 @@ import (
 	"github.com/tetratelabs/wazero/wasm/internal/wazeroir"
 )
 
+func newArchContext() archContext {
+	return archContext{
+		minimum32BitSignedInt: math.MinInt32,
+		minimum64BitSignedInt: math.MinInt64,
+	}
+}
+
 // archContext is embedded in Engine in order to store architecture-specific data.
 type archContext struct {
 	// jitCallReturnAddress holds the absolute return address for jitcall.
 	// The value is set whenever jitcall is executed and done in jit_arm64.s
 	// Native code can return back to the engine.exec's main loop back by
 	// executing "ret" instruction with this value. See arm64Compiler.exit.
 	jitCallReturnAddress uint64
+
+	// Loading large constants in arm64 is a bit costly so we place the following
```

Comment:
I guess "costly" means the fact arm64 doesn't have a straight-forward way to move a 32-bit/64-bit immediate to a register in general. On the other hand, because the constants are minimum int32 (1 << 31) and minimum int64 (1 << 63), isn't it possible to move `1` to a register and shift it instead of reading memory?

Reply:
oh  yeah lemme do some benchmarks on it

Reply:
So a bit surprisingly but the benchmark shows that

```
MOV tmRegister, $1
CMP dividend, (tmpRegister LSL #63)
```

is slower than the const load by using this constant location.

```
BenchmarkAAA
BenchmarkAAA/const_load_32-bit
BenchmarkAAA/const_load_32-bit-10               11085193               101.9 ns/op           240 B/op          4 allocs/op
BenchmarkAAA/const_load_64-bit
BenchmarkAAA/const_load_64-bit-10               11745105               102.8 ns/op           240 B/op          4 allocs/op
BenchmarkAAA/load_shift_32-bit
BenchmarkAAA/load_shift_32-bit-10               11373872               103.4 ns/op           240 B/op          4 allocs/op
BenchmarkAAA/load_shift_64-bit
BenchmarkAAA/load_shift_64-bit-10               11382826               104.3 ns/op           240 B/op          4 allocs/op
PASS
```

so I will leave it as-is. Anyway thanks!

Reply:
maybe add a footnote what you mean by costly wrt "is slower than the const load by using this constant location."

Reply:
added!